ui/app/mixins/with-component-visibility-detection.js

```diff
@@ -7,11 +8,15 @@ export default Mixin.create({
   },
 
   setupDocumentVisibility: function() {
-    this.set('_visibilityHandler', this.get('visibilityHandler').bind(this));
-    document.addEventListener('visibilitychange', this.get('_visibilityHandler'));
+    if (!Ember.testing) {
+      this.set('_visibilityHandler', this.get('visibilityHandler').bind(this));
+      document.addEventListener('visibilitychange', this.get('_visibilityHandler'));
+    }
```

Comment:
Is this similar to that on click outside not cleaning up thing I had during testing?

Reply:
This had to do with requests canceling and resulting in runloop errors being thrown in tests when I was multitasking.